hazelcast/src/main/java/com/hazelcast/spi/impl/packetdispatcher/impl/PacketDispatcherImpl.java

```diff
@@ -25,42 +25,51 @@
 import static com.hazelcast.nio.Packet.FLAG_BIND;
 import static com.hazelcast.nio.Packet.FLAG_EVENT;
 import static com.hazelcast.nio.Packet.FLAG_OP;
+import static com.hazelcast.nio.Packet.FLAG_RESPONSE;
 
 /**
  * Default {@link PacketDispatcher} implementation.
  */
-public class PacketDispatcherImpl implements PacketDispatcher {
+public final class PacketDispatcherImpl implements PacketDispatcher {
 
     private final ILogger logger;
     private final PacketHandler eventPacketHandler;
-    private final PacketHandler operationPacketHandler;
+    private final PacketHandler operationExecutor;
     private final PacketHandler connectionPacketHandler;
+    private final PacketHandler responseHandler;
 
     public PacketDispatcherImpl(ILogger logger,
-                                PacketHandler operationPacketHandler,
+                                PacketHandler operationExecutor,
+                                PacketHandler responseHandler,
                                 PacketHandler eventPacketHandler,
                                 PacketHandler connectionPacketHandler) {
         this.logger = logger;
-        this.operationPacketHandler = operationPacketHandler;
+        this.responseHandler = responseHandler;
         this.eventPacketHandler = eventPacketHandler;
         this.connectionPacketHandler = connectionPacketHandler;
+        this.operationExecutor = operationExecutor;
     }
 
     @Override
     public void dispatch(Packet packet) {
         try {
-            if (packet.isFlagSet(FLAG_OP)) {
-                operationPacketHandler.handle(packet);
-            } else if (packet.isFlagSet(FLAG_EVENT)) {
+            int flags = packet.getFlags();
+            if ((flags & FLAG_OP) != 0) {
```

Comment:
I case of doubt I look at Aeron/Agrona:

This comes from Aeron controlled fragment assembler:

```
    public Action onFragment(final DirectBuffer buffer, final int offset, final int length, final Header header)
    {
        final byte flags = header.flags();

        Action action = Action.CONTINUE;

        if ((flags & UNFRAGMENTED) == UNFRAGMENTED)
        {
            action = delegate.onFragment(buffer, offset, length, header);
        }
        else
        {
            if ((flags & BEGIN_FRAG_FLAG) == BEGIN_FRAG_FLAG)
            {
                final BufferBuilder builder = builderBySessionIdMap.computeIfAbsent(header.sessionId(), builderFunc);
                builder.reset().append(buffer, offset, length);
            }
            else
```